gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -2113,6 +2113,96 @@ gst_tensor_filter_common_get_property (GstTensorFilterPrivate * priv,
   return TRUE;
 }
 
+/**
+ * @brief Configure output tensor info with combi option.
+ */
+gboolean
+gst_tensor_filter_common_get_combined_info (GstTensorFilterPrivate * priv,
+    const GstTensorsInfo * in, const GstTensorsInfo * out,
+    GstTensorsInfo * combined)
+{
+  GList *list;
+  guint i, idx = 0;
+
+  g_return_val_if_fail (in != NULL, FALSE);
+  g_return_val_if_fail (out != NULL, FALSE);
+  g_return_val_if_fail (combined != NULL, FALSE);
+
+  gst_tensors_info_init (combined);
+
+  if (priv->combi.out_combi_i_defined || priv->combi.out_combi_o_defined) {
+    if (priv->combi.out_combi_i_defined) {
+      for (list = priv->combi.out_combi_i; list != NULL; list = list->next) {
+        i = GPOINTER_TO_INT (list->data);
+
+        if (i >= in->num_tensors) {
+          nns_loge ("Invalid input index %u, failed to combine info.", i);
+          goto error;
+        }
+
+        gst_tensor_info_copy (&combined->info[idx++], &in->info[i]);
+      }
+    }
+
+    if (priv->combi.out_combi_o_defined) {
+      for (list = priv->combi.out_combi_o; list != NULL; list = list->next) {
+        i = GPOINTER_TO_INT (list->data);
+
+        if (i >= out->num_tensors) {
+          nns_loge ("Invalid output index %u, failed to combine info.", i);
+          goto error;
+        }
+
+        gst_tensor_info_copy (&combined->info[idx++], &out->info[i]);
+      }
+    }
+
+    combined->num_tensors = idx;
+  } else {
+    gst_tensors_info_copy (combined, out);
+  }
+
+  return TRUE;
+
+error:
+  gst_tensors_info_init (combined);
```

Comment:
A tensor name of the combined info is resource leak.
```suggestion
  gst_tensors_info_free (combined);
```

Reply:
updated